### PR TITLE
feat: Report usage of stitchAsyncCode

### LINKED ISSUE #1944: Track how many people use stitchAsyncCode
### Description

We currently have no clue how many people are using the experimental feature stitchAsyncCode. We should find a way to track that in Looker.

### PR DESCRIPTION
**Pull request by kevinrenskers**
## :scroll: Description

We can now report usage of the stitchAsyncCode option, as an integration.

## :bulb: Motivation and Context

Closes #1944.

## :green_heart: How did you test it?

Unit test.

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [x]  I updated the docs if needed
- [x] Review from the native team if needed
- [x] No breaking changes

### CODE REVIEW

**Comment by kevinrenskers** [comment]
One question about the `SentryClient.setSdk` method:

```
- (void)setSdk:(SentryEvent *)event
{
    if (event.sdk) {
        return;
    }

    id integrations = event.extra[@"__sentry_sdk_integrations"];
    if (!integrations) {
        integrations = [NSMutableArray new];
        for (NSString *integration in SentrySDK.currentHub.installedIntegrationNames) {
            // Every integration starts with "Sentry" and ends with "Integration". To keep the
            // payload of the event small we remove both.
            NSString *withoutSentry = [integration stringByReplacingOccurrencesOfString:@"Sentry"
                                                                             withString:@""];
            NSString *trimmed = [withoutSentry stringByReplacingOccurrencesOfString:@"Integration"
                                                                         withString:@""];
            [integrations addObject:trimmed];
        }

        if (self.options.stitchAsyncCode) {
            [integrations addObject:@"StitchAsyncCode"];
        }
    }

    event.sdk = @{
        @"name" : SentryMeta.sdkName,
        @"version" : SentryMeta.versionString,
        @"integrations" : integrations
    };
}
```

This line `id integrations = event.extra[@"__sentry_sdk_integrations"];`.. I can't find the string `__sentry_sdk_integrations` anywhere in the project. Is this for hybrid SDKs so they can set their own "secret" integrations string on an event? Or is this old code that should be removed?

File: CHANGELOG.md
Comment (armcknight) [inline CHANGELOG.md:3]: Unrelated to these changes, but I find it annoying to have to keep re-adding this after each release. I've filed an issue here proposing improvements, please check it out and +1 it if you agree! https://github.com/getsentry/craft/issues/422

**Comment by armcknight** [comment]
> One question about the `SentryClient.setSdk` method:
> ...
> This line `id integrations = event.extra[@"__sentry_sdk_integrations"];`.. I can't find the string `__sentry_sdk_integrations` anywhere in the project. Is this for hybrid SDKs so they can set their own "secret" integrations string on an event? Or is this old code that should be removed?

It appears in the Unity repo, in an old submodule version of this repo, I guess we removed it from here but Unity is using an old version (7.13.0, we're now on 7.28.0): https://github.com/getsentry/sentry-cocoa/blob/2f02b39a9a47581830d6c0c92d07f5d0959bc221/Sources/Sentry/SentryClient.m#L571-L572

Not sure if it's even used there ¯\\_(ツ)_/¯ My grep would've also turned up hits in our docs, but it's not there either.

**Comment by kevinrenskers** [comment]
Why is this PR not getting any reviews?

**Review by brustolin — APPROVED** [review]

**Comment by armcknight** [comment]
@kevinrenskers I see you were asking for some reviews on this and https://github.com/getsentry/sentry-cocoa/pull/2263. Personally, I don't review PRs that have failing tests, to avoid review churn. Why were these PRs merged with the VLC and Home Assistant integration tests in a failure state?

**Comment by kevinrenskers** [comment]
Both PRs had green checks actually. 

After they were approved I had to fix the changelog conflict, and immediately merged without waiting for tests. After all, I only changed the changelog.

But yea they were both waiting for review with green checks.

**Comment by armcknight** [comment]
Ah yeah I get you, that's annoying. There's a lot to say there about github's triggers, I've previously filed a support ticket for that and they said it works as intended 🙄

**Pull request merged**
